### PR TITLE
Zephyr 64-bit port: initial set of changes (v2)

### PR DESCRIPTION
**Pull request by npitre**
This is my first set of patches containing very simple changes to start
with. Those should be pretty uncontrovertial. Also included is a genuine
bugfix for the PTP test code. Much more is coming after this.

Changes from v1:
Tweaks to please gitlint.
I'm not line-wrapping compiler output quotes though.

### CODE REVIEW

**Review by andyross — APPROVED** [review]
As mentioned on the previous patch: code changes all look great to me.  Some fighting left to do with gitlint it seems.

**Comment by pfalcon** [comment]
@andyross: Maybe you could give a hint how this relates to x86_64 port done by you previously?

File: arch/posix/core/posix_core.c [hunk 1]
```diff
@@ -479,9 +479,7 @@ void z_impl_k_thread_abort(k_tid_t thread)
 	unsigned int key;
 	int thread_idx;
 
-	posix_thread_status_t *tstatus =
-					(posix_thread_status_t *)
-					thread->callee_saved.thread_status;
+	posix_thread_status_t *tstatus = thread->callee_saved.thread_status;
```
Comment (aescolar) [inline arch/posix/core/posix_core.c]: This commit title would be misleading: You are indeed removing here a relatively unnecessary cast ( void * =>posix_thread_status_t * ). But it is only unnecessary after you have changed the type of thread_status.
Could you please change the commit description to mention instead that you are changing the type of a variable which was used as pointer storage from uint32 to void* ?

**Review by aescolar — CHANGES_REQUESTED** [review]
One nit by now (I need to leave now, will come back to it in a couple of hours. So holding merge until I can review it)

**Review by tbursztyka — APPROVED** [review]
Fine by me.

However @npitre, do not close a PR, and push a new one. Just git push -f your branch. Same PR will be updated. It's much easier to follow comments/changes then.

**Comment by aescolar** [comment]
@npitre 2 minor procedural comments:
* As @tbursztyka commented, you do no need to close a pull request (PR) to update it. You can just push the changes to the branch in your GitHub clone (the one you are using as origin of this PR), and the PR will be updated. Doing so will be a bit easier for others to follow, as comments and reviews are not lost. (Note that I just linked the other PR into this one)
* This PR contains changes that relate to quite different areas maintained by different people. And the patches are not directly dependent on each other. In the future, in cases like this, you may want to create different PRs, one for each area, as that will probably simplify communication and reviewing.

File: arch/posix/core/posix_core.c (hunk 1 above)
Comment (aescolar) [reply arch/posix/core/posix_core.c]: Anyhow, I guess this cast did not really harm there even after being from a (void *), or?

File: boards/posix/native_posix/irq_ctrl.h [hunk 2]
```diff
@@ -26,7 +26,7 @@ uint8_t hw_irq_ctrl_get_prio(unsigned int irq);
 int hw_irq_ctrl_get_highest_prio_irq(void);
 uint32_t hw_irq_ctrl_get_current_lock(void);
 uint32_t hw_irq_ctrl_change_lock(uint32_t new_lock);
-uint64_t hw_irq_ctrl_get_irq_status(void);
+u64_t hw_irq_ctrl_get_irq_status(void);
```
Comment (aescolar) [inline boards/posix/native_posix/irq_ctrl.h]: wouldn't a zephyr/types.h include be missing for this? (to keep this header self-contained)

**Review by aescolar — DISMISSED** [review]
Just 2 nit-picks from me. (The commit message and the missing header include), otherwise this seems fine for the posix arch/native_posix side.

**Comment by npitre** [comment]
On Wed, 29 May 2019, Alberto Escolar wrote:

> -	posix_thread_status_t *tstatus =
> -					(posix_thread_status_t *)
> -					thread->callee_saved.thread_status;
> +	posix_thread_status_t *tstatus = thread->callee_saved.thread_status;
> 
> Anyhow, I guess this cast did not really harm there even after being from a (void *), or?

Any cast is a potential source of trouble. They completely shortcircuit 
type checking. At least without the cast, the compiler expects a pointer 
and will complain if it doesn't get one. Using void* isn't perfect 
either but at least there won't be any mistake from an integer type.

To be even cleaner I could have added a forward declaration for 
posix_thread_status_t in arch/posix/include/kernel_arch_thread.h, but
that wouldn't work with a typedef, and I imagine there is a reason why
this is currently defined separately in arch/posix/include/posix_core.h.
Hence the void pointer.

**Comment by npitre** [comment]
On Wed, 29 May 2019, Alberto Escolar wrote:

> -uint64_t hw_irq_ctrl_get_irq_status(void);
> +u64_t hw_irq_ctrl_get_irq_status(void);
> 
> wouldn't a zephyr/types.h include be missing for this? (to keep this header self-contained)

No. It is a plain definition vs declaration mismatch. Both have to agree 
even if the type bouls down to the same thing.

**Comment by npitre** [comment]
On Wed, 29 May 2019, Alberto Escolar wrote:

> As @tbursztyka commented, you do no need to close a pull request 
> (PR) to update it.

OK.

> This PR contains changes that relate to quite different areas 
> maintained by different people. And the patches are not directly 
> dependent on each other. In the future, in cases like this, you may 
> want to create different PRs, one for each area, as that will probably 
> simplify communication and reviewing.

Well... I have even more wide-ranging changes coming up. This PR is only 
the super trivial ones which didn't warrant a dedicated PR of their own 
(or so I thought). Future changes have a more focussed purpose, but some 
of them touch many files and splitting them up might get impractical. I 
hope you'll indulge me a bit.

**Comment by npitre** [comment]
On Tue, 28 May 2019, Paul Sokolovsky wrote:

> @andyross: Maybe you could give a hint how this relates to x86_64 port done by you previously?

My understanding is that the x86_64 port is using the x32 ABI meaning 
that it is not truly 64 bits per se.

**Comment by pfalcon** [comment]
> My understanding is that the x86_64 port is using the x32 ABI meaning that it is not truly 64 bits per se.

Thanks @npitre, that explains the situation, and why there're still cases when we use `int` where `void*` should be a better choice (and I'm not even talking about uintptr_t), and nobody complains ;-).

**Comment by andyross** [comment]
That's exactly it: the x86_64 architecture support is or a 64 bit machine, but runs the compiler in a mode where pointers are still 32 bit (because of precisely these kinds of word size bugs in the Zephyr kernel).  The bottom layer is ABI agnostic, and in theory we could just remove the -mx32 compiler flag to exercise it there too.

But it's much easier with the posix layer, so I'm very happy to see it done here first. :)

**Comment by npitre** [comment]
On Wed, 29 May 2019, Andy Ross wrote:

> But it's much easier with the posix layer, so I'm very happy to see it done here first. :)

And to be clear, I do have it working with native_posix and passing the 
test suite here already. The set of needed changes are way more 
extensive than what I've published so far though.

**Comment by npitre** [comment]
On Wed, 29 May 2019, Tomasz Bursztyka wrote:

> However @npitre, do not close a PR, and push a new one. Just git push -f your branch. Same PR will be updated.

I did just that, removing the PTP fix for which I did a separate PR, and 
the native_posix changes still under discussion that I'll join to other 
posix changes later on.

**Pull request merged**